code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,430 @@
+		/obj/item/clothing/gloves/krav_maga,
+		/obj/item/nullrod,
+		/obj/item/tank/jetpack,
+		/obj/item/documents,
+		/obj/item/nuke_core_container
+	)
+	// These items will NOT be preserved
+	var/static/list/do_not_preserve_items = list (
+		/obj/item/mmi/posibrain
+	)
+
+/obj/machinery/cryopod/Initialize()
+	..()
+	return INITIALIZE_HINT_LATELOAD //Gotta populate the cryopod computer GLOB first
+
+/obj/machinery/cryopod/LateInitialize()
+	update_icon()
+	find_control_computer()
+
+// This is not a good situation
+/obj/machinery/cryopod/Destroy()
+	control_computer = null
+	return ..()
+
+/obj/machinery/cryopod/proc/find_control_computer(urgent = 0)
+	for(var/M in GLOB.cryopod_computers)
+		var/obj/machinery/computer/cryopod/C = M
+		if(get_area(C) == get_area(src))
+			control_computer = C
+			break
+
+	// Don't send messages unless we *need* the computer, and less than five minutes have passed since last time we messaged
+	if(!control_computer && urgent && last_no_computer_message + 5*60*10 < world.time)
+		log_admin("Cryopod in [get_area(src)] could not find control computer!")
+		message_admins("Cryopod in [get_area(src)] could not find control computer!")
+		last_no_computer_message = world.time
+
+	return control_computer != null
+
+/obj/machinery/cryopod/close_machine(mob/user)
+	if(!control_computer)
+		find_control_computer(TRUE)
+	if((isnull(user) || istype(user)) && state_open && !panel_open)
+		..(user)
+		var/mob/living/mob_occupant = occupant
+		if(mob_occupant && mob_occupant.stat != DEAD)
+			to_chat(occupant, "<span class='boldnotice'>You feel cool air surround you. You go numb as your senses turn inward.</span>")
+		if(mob_occupant.client) // if they're logged in
+			despawn_world_time = world.time + (time_till_despawn * 0.1) // This gives them 30 seconds
+		else
+			despawn_world_time = world.time + time_till_despawn
+	icon_state = "cryopod"
+
+/obj/machinery/cryopod/open_machine()
+	..()
+	icon_state = "cryopod-open"
+	density = TRUE
+	name = initial(name)
+
+/obj/machinery/cryopod/container_resist_act(mob/living/user)
+	visible_message("<span class='notice'>[occupant] emerges from [src]!</span>",
+		"<span class='notice'>You climb out of [src]!</span>")
+	open_machine()
+
+/obj/machinery/cryopod/relaymove(mob/user)
+	container_resist_act(user)
+
+/obj/machinery/cryopod/process()
+	if(!occupant)
+		return
+
+	var/mob/living/mob_occupant = occupant
+	if(mob_occupant)
+		// Eject dead people
+		if(mob_occupant.stat == DEAD)
+			open_machine()
+
+		if(!(world.time > despawn_world_time))
+			return
+
+		if(!mob_occupant.client && mob_occupant.stat < 2) // Occupant is living and has no client.
+			if(!control_computer)
+				find_control_computer(urgent = TRUE) // better hope you found it this time
+
+			despawn_occupant()
+
+/obj/machinery/cryopod/proc/handle_objectives()
+	var/mob/living/mob_occupant = occupant
+	// Update any existing objectives involving this mob.
+	for(var/datum/objective/O in GLOB.objectives)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

Reply:
here and everywhere else